skbio/stats/composition.py

```diff
@@ -608,10 +611,342 @@ def centralize(mat):
+
+    >>> grouping = pd.Series([0, 0, 0, 1, 1, 1],
+    ...                      index=['s1','s2','s3','s4','s5','s6'])
+
+    Now run `ancom` and see if there are any features that have any
+    significant differences between the treatment and the control.
+
+    >>> results = ancom(table, grouping)
+    >>> results['W']
+    b1    0
+    b2    4
+    b3    1
+    b4    1
+    b5    1
+    b6    0
+    b7    1
+    Name: W, dtype: int64
+
+    The W-statistic is the number of features that a single feature is tested
+    to be significantly different against.  In this scenario, b2 was detected
+    to have significantly different abundances compared to four of the other
+    species. To summarize the results from the W-statistic, let's take a look
+    at the results from the hypothesis test:
+
+    >>> results['reject']
+    b1    False
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    Name: reject, dtype: bool
+
+    From this we can conclude that only b2 was significantly
+    different between the treatment and the control.
+
+    """
+
+    if not isinstance(table, pd.DataFrame):
+        raise TypeError('`table` must be a `pd.DataFrame`, '
+                        'not %r' % type(table).__name__)
+    if not isinstance(grouping, pd.Series):
+        raise TypeError('`grouping` must be a `pd.Series`,'
+                        ' not %r' % type(grouping).__name__)
+
+    table_index_len = len(table.index)
+    grouping_index_len = len(grouping.index)
+    table, grouping = table.align(grouping, axis=0, join='inner')
+    if (len(table) != table_index_len or len(grouping) != grouping_index_len):
+        raise ValueError('`table` index and `grouping` '
+                         'index must be consistent.')
+
+    if np.any(table == 0):
+        raise ValueError('Cannot handle zeros or negative values in `table`. '
+                         'Use pseudo counts or `multiplicative_replacement`.')
+
+    if not 0 < alpha < 1:
+        raise ValueError('alpha=%f is not within 0 and 1' % alpha)
+
+    if not 0 < tau < 1:
+        raise ValueError('tau=%f is not within 0 and 1' % tau)
+
+    if not 0 < theta < 1:
+        raise ValueError('theta=%f is not within 0 and 1' % theta)
+
+    if multiple_comparisons_correction is not None:
+        if multiple_comparisons_correction != 'holm-bonferroni':
+            raise ValueError('%r is not an option for '
+                             '`multiple_comparisons_correction`'
+                             % multiple_comparisons_correction)
+
+    if (grouping.isnull()).any():
+        raise ValueError('Cannot handle missing values in `grouping`.')
+
+    if (table.isnull()).any().any():
+        raise ValueError('Cannot handle missing values in `table`.')
+
+    groups, _grouping = np.unique(grouping, return_inverse=True)
+    grouping = pd.Series(_grouping, index=grouping.index)
+    num_groups = len(groups)
+    if len(grouping) != len(table):
```

Comment:
This check is redundant right? Also the error message doesn't make sense, it was copied from anosim.